packages/docusaurus-theme-classic/src/theme/MDXComponents/index.tsx

```diff
@@ -14,17 +14,28 @@ import type {MDXComponentsObject} from '@theme/MDXComponents';
 const MDXComponents: MDXComponentsObject = {
   code: (props) => {
     const {children} = props;
-    if (typeof children === 'string') {
```

Comment:
This line of code causes current bug because HTML entities come as an array of strings, not a regular string.